Jeremy Huntwork reported:

on a livecd, hotplug loads the tulip driver for my network card:

tulip0:  EEPROM default media type Autosense.
tulip0:  Index #0 - Media MII (#11) described by a 21140 MII PHY (1) block.
tulip0:  Index #1 - Media 10baseT (#0) described by a 21140 non-MII (0) block.
tulip0:  Index #2 - Media 100baseTx (#3) described by a 21140 non-MII (0) block.
tulip0:  Index #3 - Media 10baseT-FDX (#4) described by a 21140 non-MII (0) block.
tulip0:  Index #4 - Media 100baseTx-FDX (#5) described by a 21140 non-MII (0) block.
tulip0:  MII transceiver #1 config 3100 status 7809 advertising 01e1.
eth0: Davicom DM9102/DM9102A rev 49 at 0001ec00, 00:80:AD:D0:6A:C4, IRQ 18.
eth0: Setting full-duplex based on MII#1 link partner capability of 45e1.

However, after some time, the card stops working - the network connection sort
of freezes. If I 'rmmod tulip' and then 'modprobe dmfe' and restart the
network, everything works and is stable.

dmfe: Davicom DM9xxx net driver, version 1.36.4 (2002-01-17)
ACPI: PCI Interrupt 0000:02:09.0[A] -> GSI 18 (level, low) -> IRQ 18
eth0: Davicom DM9102 at pci0000:02:09.0, 00:80:ad:d0:6a:c4, irq 18.

Signed-off-by: Alexander E. Patrakov

--- linux-2.6.15-rc1/drivers/net/tulip/tulip_core.c	2006-02-21 17:22:47.000000000 +0500
+++ linux-2.6.15-rc1/drivers/net/tulip/tulip_core.c	2006-02-21 17:23:35.000000000 +0500
@@ -223,8 +223,8 @@
 	{ 0x1259, 0xa120, PCI_ANY_ID, PCI_ANY_ID, 0, 0, COMET },
 	{ 0x11F6, 0x9881, PCI_ANY_ID, PCI_ANY_ID, 0, 0, COMPEX9881 },
 	{ 0x8086, 0x0039, PCI_ANY_ID, PCI_ANY_ID, 0, 0, I21145 },
-	{ 0x1282, 0x9100, PCI_ANY_ID, PCI_ANY_ID, 0, 0, DM910X },
-	{ 0x1282, 0x9102, PCI_ANY_ID, PCI_ANY_ID, 0, 0, DM910X },
+/*	{ 0x1282, 0x9100, PCI_ANY_ID, PCI_ANY_ID, 0, 0, DM910X }, */
+/*	{ 0x1282, 0x9102, PCI_ANY_ID, PCI_ANY_ID, 0, 0, DM910X }, */
 	{ 0x1113, 0x1216, PCI_ANY_ID, PCI_ANY_ID, 0, 0, COMET },
 	{ 0x1113, 0x1217, PCI_ANY_ID, PCI_ANY_ID, 0, 0, MX98715 },
 	{ 0x1113, 0x9511, PCI_ANY_ID, PCI_ANY_ID, 0, 0, COMET },
